shell/browser/ui/cocoa/electron_ns_window.mm

```diff
@@ -163,12 +162,6 @@ - (NSView*)frameView {
   return [[self contentView] superview];
 }
 
-// By overriding this built-in method the corners of the vibrant view (if set)
-// will be smooth.
-- (NSImage*)_cornerMask {
-  return [self cornerMask];
```

Comment:
I'm not sure if removing this would cause side effects on older macOS versions, could something like this solve the problem?

```objective-c
- (NSImage*)_cornerMask {
  if ([self hasVibrantView])
    return [self cornerMask];
  else
    return [super _cornerMask];
}
```

Reply:
yea i'm not sure as well what could this cause on older macOS versions, but calling `[super _cornerMask]` for me results in an error:

```
../../electron/shell/browser/ui/cocoa/electron_ns_window.mm:182:17: error: 'NativeWidgetMacNSWindow' may not respond to '_cornerMask' [-Werror]
  return [super _cornerMask];
          ~~~~~ ^
1 error generated.
```

and if i just return `nil` always in that method the resize has the same lagginess.

Reply:
You can declare the API like this:

```objective-c
@interface NSWindow (PrivateAPI)
- (NSImage*)_cornerMask;
@end
```

Reply:
with this it works, so we could do that above, the question is, having smooth corners are better than not having poor resize performance in vibrancy mode?

Reply:
I have tried removing `_cornerMask` on 10.14 and the corners are still smooth, and looking at original issue https://github.com/electron/electron/issues/10886 this might only be a problem for macOS <= 10.13.

So how about only doing `_cornerMask` hack when macOS version <= 10.13?

Reply:
actually I realised now that if I only return our cornerMask for vibrant view, the resize performance is the same as calling the base class implementation, so possibly it will work well on all platforms this way